### PR TITLE
Allow a list of schemas when switching using schemas

### PR DESCRIPTION
**Pull request by marksiemers**

### CODE REVIEW

**Comment by marksiemers** [comment]
Our team is looking to make a switch to this fork of apartment, and we need to continue supporting multiple schemas for a single tenant (optionally).

We have been running this in production for about 2 years. I've made a modification (switch from ternary to an `Array.wrap`) - I can switch that back or make any other changes that would make this acceptable to merge.

What do you all think?

**Comment by rpbaltazar** [comment]
Codewise looks like it would work without major breaking changes.
I have not yet fully understood what is the use case we are trying to solve with this.

**Comment by marksiemers** [comment]
@rpbaltazar - For our customers, we import a lot of data (about 5 TB across 1,900 customers each night).

We've had to do a lot of things to get this import process to complete in under 12 hours and also have the customer dashboards available 24/7.

One of the things we do is put most of the data into an "import schema" which lives during the import process for a customer, and is removed once we are done.

For tables where most or all of the data will change, we create that table in the import schema, and do the work there. For tables where most of the data will stay the same, we do NOT copy the tables/data over - we mark rows as "inactive" or "marked for deletion" until the import process is finished.

An example of what this might look like:
During and import process we have the following schemas for a customer:
```ruby
'customer1' # The "live" schema for the customer
'customer1-2020-10-02-6a87fe' # The "import" schema for the customer
```
In the live schema, we have the following tables:
```
table_a
table_b
table_c
```
We detect during the import process that:
- 100% of table_a's rows will change
- 5% of table_b's rows will change
- 0% of table_c's rows will change

This means that during the import, the import schema will only have `table_a` and it will not have b or c.
We want to prioritize the import schema (`customer1-2020-10-02-6a87fe`) so that when we reference `table_a` it will use that table in the import schema (not the "live" schema), but for table_b and table_c, since they don't exist in the import schema, we want it to fall-back to the "live" schema.

I hope I've explained this well - it is a complex system and it took us a while to get to this point, but it has been necessary to meet our customers' needs.

**Comment by marksiemers** [comment]
@rpbaltazar - WDYT?

We're doing some work now that will require us to update/upgrade the apartment gem.

If this isn't ready to be merged in, then we'll create a fork of the repo and apply these changes in our fork.

We would rather use the official gem/release, but understand if this use-case isn't applicable broadly enough to include in the gem.

File: lib/apartment/adapters/postgresql_adapter.rb [hunk 1]
```diff
@@ -133,6 +131,12 @@ def reset_sequence_names
           c.remove_instance_variable :@sequence_name if c.instance_variable_defined?(:@sequence_name)
         end
       end
+
+      def schema_exists?(schemas)
```
Comment (rpbaltazar) [inline lib/apartment/adapters/postgresql_adapter.rb:135]: This method is very similar to `tenant_exists`

**Review by rpbaltazar — CHANGES_REQUESTED** [review]
small change suggested

**Comment by rpbaltazar** [comment]
> @rpbaltazar - WDYT?
> 
> We're doing some work now that will require us to update/upgrade the apartment gem.
> 
> If this isn't ready to be merged in, then we'll create a fork of the repo and apply these changes in our fork.
> 
> We would rather use the official gem/release, but understand if this use-case isn't applicable broadly enough to include in the gem.

Thank you for the PR submission and apologies for my absence in this repo. I've been quite restricted in terms of free time to dedicate to the OSS.

Thank you also for the detailed description of your use case. Seems indeed a rather complex solution but i think you've explained it quite well.

**Comment by rpbaltazar** [comment]
@marksiemers do you mind syncing the branch so i can merge this, please?

**Comment by citrus** [comment]
+1 for this feature!

**Comment by marksiemers** [comment]
@rpbaltazar - I've tried to get this cleaned up, but haven't been able to update this PR nor open a new one. When I do a force push after rebasing, nothing new shows up here. When I try to open the PR, this repo is not available (the influitive one and hundreds of others are available)

Perhaps when I first opened this, this repo was still a fork of the influitive repo and now it isn't?

I think what I'll try next is a fresh clone of this repo, and created a PR from my personal github org, and see if that does the trick

**Comment by marksiemers** [comment]
Closing in favor of this PR: https://github.com/rails-on-services/apartment/pull/154

**Pull request closed without merging**